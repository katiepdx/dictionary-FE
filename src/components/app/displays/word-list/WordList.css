.wordItem {
  border: 2px solid black;
  font-size: 1.2rem;
  text-align: center;
}

.wordItem:hover {
  background-color: wheat;
  transition: .6s;
}
