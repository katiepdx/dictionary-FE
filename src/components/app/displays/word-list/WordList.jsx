import React from 'react'
import WordItem from '../word-item/WordItem'
import PropTypes from 'prop-types'
import styles from './WordList.css'

const WordList = ({ wordsList }) => {
  return wordsList.map(word => (
    <div key={word.id} className={styles.wordItem}>
      <WordItem {...word} />
    </div>
  ))
}

WordList.propTypes = {
  wordsList: PropTypes.arrayOf(PropTypes.shape({
    word: PropTypes.string.isRequired,
    wordLanguage: PropTypes.string.isRequired,
    wordTranslation: PropTypes.string,
    wordDefinition: PropTypes.string.isRequired,
    exampleSentence: PropTypes.string
  }))
}

export default WordList
