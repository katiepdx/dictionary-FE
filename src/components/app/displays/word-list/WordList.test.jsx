import React from 'react'
import { render } from '@testing-library/react'
import WordList from './WordList'

const list = [
  {
    word: "Word 1",
    wordLanguage: "English",
    wordTranslation: "word 1 translation",
    wordDefinition: "word 1 definition",
    exampleSentence: "this is a cool word",
    notes: "I learned a new word!",
  }, {
    word: "Word 2",
    wordLanguage: "English",
    wordTranslation: "word 2 translation",
    wordDefinition: "word 2 definition",
    exampleSentence: "this is a cool word",
    notes: "I learned a new word!",
  }
]

describe('<WordList/>', () => {
  it('should render a list of words to the page', () => {
    const { asFragment } = render(<WordList wordsList={list} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
