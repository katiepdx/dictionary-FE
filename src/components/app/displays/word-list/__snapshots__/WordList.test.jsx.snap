// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordList/> should render a list of words to the page 1`] = `
<DocumentFragment>
  <div
    class="wordItem"
  >
    <div
      class="wordItem"
    >
      <h4>
        Word: Word 1
      </h4>
      <p
        class="label"
      >
        Language
      </p>
      <p>
         English
      </p>
      <p
        class="label"
      >
        Translation
      </p>
      <p>
         word 1 translation
      </p>
      <p
        class="label"
      >
        Definition
      </p>
      <p>
         word 1 definition
      </p>
      <p
        class="label"
      >
        Example Sentence
      </p>
      <p>
         this is a cool word
      </p>
      <p
        class="label"
      >
        Notes
      </p>
      <p>
         I learned a new word!
      </p>
    </div>
  </div>
  <div
    class="wordItem"
  >
    <div
      class="wordItem"
    >
      <h4>
        Word: Word 2
      </h4>
      <p
        class="label"
      >
        Language
      </p>
      <p>
         English
      </p>
      <p
        class="label"
      >
        Translation
      </p>
      <p>
         word 2 translation
      </p>
      <p
        class="label"
      >
        Definition
      </p>
      <p>
         word 2 definition
      </p>
      <p
        class="label"
      >
        Example Sentence
      </p>
      <p>
         this is a cool word
      </p>
      <p
        class="label"
      >
        Notes
      </p>
      <p>
         I learned a new word!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
